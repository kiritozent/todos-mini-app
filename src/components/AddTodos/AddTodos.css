.FormItem {
  padding: 0px;
  margin: 0px;
  position: absolute;
  right: 16px;
  width: 400px;
  max-width: calc(100% - 80px);
  top: 5.5px;
}
