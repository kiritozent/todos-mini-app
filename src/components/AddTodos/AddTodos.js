import React, { useState, useRef, useEffect } from 'react';
import { Input, Form } from 'antd';
import PropTypes from 'prop-types';
import { FileAddOutlined } from '@ant-design/icons';
import './AddTodos.css';
import { DEFAULT_REDUCERS } from '../../data/const';

const AddTodos = props => {
  const { postTodos, postTodosRequest } = props;
  const addTodosInputRef = useRef();
  const [todosText, setTodosText] = useState();

  useEffect(() => {
    if (postTodos.payload) {
      setTodosText('');
    }
  }, [postTodos.payload]);

  function onAddTodos() {
    const user = JSON.parse(localStorage.getItem('user'));
    const data = {
      title: todosText,
      completed: false,
      userId: user.id,
    };
    postTodosRequest(data);
  }

  return (
    <Form.Item
      className="FormItem"
      validateStatus="validating"
      hasFeedback={postTodos.fetching}
    >
      <Input
        ref={addTodosInputRef}
        value={todosText}
        placeholder="Add new todos"
        onChange={e => setTodosText(e.target.value)}
        onKeyPress={e => {
          if (e.key === 'Enter') {
            onAddTodos();
          }
        }}
        disabled={postTodos.fetching}
        prefix={<FileAddOutlined />}
      />
    </Form.Item>
  );
};

AddTodos.propTypes = {
  postTodos: PropTypes.object,
  postTodosRequest: PropTypes.func,
};

AddTodos.defaultProps = {
  postTodos: DEFAULT_REDUCERS,
  postTodosRequest: () => {},
};

export default AddTodos;
