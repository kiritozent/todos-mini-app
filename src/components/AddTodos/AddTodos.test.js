import React from 'react';
import { shallow } from 'enzyme';
import AddTodos from './AddTodos';

describe('AddTodos', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<AddTodos />);
    expect(testCase).toMatchSnapshot();
  });
});
