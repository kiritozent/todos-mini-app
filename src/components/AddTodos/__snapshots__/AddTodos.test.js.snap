// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodos render without crashing passed 1`] = `
<FormItem
  className="FormItem"
  hasFeedback={false}
  validateStatus="validating"
>
  <Input
    disabled={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Add new todos"
    prefix={<ForwardRef(FileAddOutlined) />}
    type="text"
  />
</FormItem>
`;
